lib/Twig/Filter.php

```diff
@@ -34,6 +34,10 @@ class Twig_Filter
      */
     public function __construct(string $name, $callable = null, array $options = array())
     {
+        if (__CLASS__ !== get_class($this)) {
+            @trigger_error('Overriding '.__CLASS__.' is deprecated since version 2.4.0 and the class will be final in 3.0.', E_USER_DEPRECATED);
```

Comment:
then, we will make it a pain for the whole Symfony community or they will just drop support for PHP 5.x in bundles to be able to avoid having to do checks on which class to use for each function and filter they expose. Have you tried to update Symfony itself to avoid these deprecations ?
Thus, the check for which class to use even has to be done by detecting the Twig version rather than the availability of features, as Twig_Filter also exists in 1.x (but is a different class, which is why we suffer all this pain btw).
At least, making them migrate to `Twig\Filter` directly would avoid doing 2 migrations (and can be detected by `class_exists). But it does not solve the issue regarding the fact that Symfony 3 LTS is stuck with Twig 1.x support